Model/Export/ConvertToXls.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace JustBetter\ProductGridExport\Model\Export;
+
+use Magento\Framework\Convert\ExcelFactory;
+use Magento\Framework\Filesystem;
+use Magento\Ui\Component\MassAction\Filter;
+use Magento\Ui\Model\Export\ConvertToXml;
+use Magento\Ui\Model\Export\MetadataProvider;
+use Magento\Ui\Model\Export\SearchResultIteratorFactory;
+
+class ConvertToXls extends ConvertToXml
+{
+
+    /**
+     * Returns XLS file
+     *
+     * @return array
+     * @throws \Magento\Framework\Exception\FileSystemException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function getXlsFile()
+    {
+        $component = $this->filter->getComponent();
+
+        $name = md5(microtime());
+        $file = 'export/'. $component->getName() . $name . '.xls';
+
+        $this->filter->applySelectionOnTargetProvider();
+        $dataProvider = $component->getContext()->getDataProvider();
+
+        // Force all results
+        $searchResult = $dataProvider->getSearchResult()->setCurPage(1)
+            ->setPageSize(0);
```

Comment:
Does this still work well with large datasets because we're fetching all products with all their data right?

Since they want an iterator we have freedom to "lazily" load the results.
I'm not sure if they accept Generators as well https://www.php.net/manual/en/language.generators.syntax.php
If they do, this should suffice
```php
function searchResultsGenerator($searchResult)
{
    $totalCount = (int) $searchResult->getSize();
    while ($totalCount > 0) {
        $items = $searchResult->getItems();
        yield from $items;

        $searchCriteria->setCurPage(++$page);
        $totalCount = $totalCount - $this->pageSize;
    }
}

$searchResultIterator = searchResultsGenerator($searchResult);
```

If that does not work we could make an iterator ourselves https://www.php.net/manual/en/language.generators.comparison.php
(or possibly use the SearchResultIterator in `\Magento\Framework\Data\SearchResultIterator`)